components/_expanders/expander.js

```diff
@@ -0,0 +1,39 @@
+const expander = {
+  init: function() {
+    this.cacheDom()
+    this.bindEvents()
+  },
+  cacheDom: function() {
+    this.expanderComponents = document.querySelectorAll('.msds-expander__list-item')
```

Comment:
if this is a single expander component we are initializing here, then we should not cache all the components here. as one component should not know about the others in this scope. it only cares about it self

Reply:
Do you have a example or could you provide a demo using this code?

Reply:
sorry for a slightly longer comment here.

I guess there are many ways you could build this component, and I think we need to figure out in the team what solution we want to go with, going forward. However, when we are building a component it needs to be encapsulated. A comment way to handle that would be in your CacheDom Method where you would do something like this. 
using the object literal pattern that your are using here.

cacheDom: function() {
   this.expanderComponent = document.querySelector(this.id);
}

And then your Init/create/constructor method would be something like  
init: function(id) {
 this.id = id
}

openExpander: function(){ Open your expander Module}
closeExpander: function(){  Close your expander Module}
Then in your initialization module, you would probably do something like 

let myExpander = expander.init("expander-1");
let mySecondExpander = expander.init("expander-2");

let expanderIntializationModule = {
    cacheDom: function() {
    this.expanderComponents = document.querySelectorAll('.msds-expander__list-item')
    },
    //bind your click event logic here. and then call detect what module has been clicked and then loop throug all your modules and call the closeExpander method on them. and the clicked module you would call the openExpander method on.
}

Reply:
I agree with Danny. Just to give you some example why defining logic of another component here is bad idea in my opinion. In the child component we can lock some specific logic related to external copmponents, so we assume that this logic always will be valid for all cases - in this specific component you always want to close another expenders when one is open. 

This is bad - you can't predict that. And this will probably need to be adjust on first usage on this component on PCT. Please notice that we are going to nest expanders inside expanders on PCT page, so this code probably will close parent expander when children one will be open - everything will be closed.

Generally managing state of children components by children components is bad design. State of children components should by manage by parent component. That's the reason I think moving logic of closing another expanders should be moved level up - to parent component.

And if you think that this logic will be used often (which is proabbly true) - closing another components - you can move it to separated 'helper' function. This function can be executed in the parent component when needed - without overwriting this functionality when is no needed.